src/features/session_replay/aggregate/index.js

```diff
@@ -68,58 +71,65 @@ export class Aggregate extends AggregateBase {
     /** A value which increments with every new mutation node reported. Resets after a harvest is sent */
     this.payloadBytesEstimation = 0
 
-    /** The method to stop recording. This defaults to a noop, but is overwritten once the recording library is imported and initialized */
-    this.stopRecording = () => { /* no-op until set by rrweb initializer */ }
-
-    // The SessionEntity class can emit a message indicating the session was cleared and reset (expiry, inactivity). This feature must abort and never resume if that occurs.
-    this.ee.on('session-reset', () => {
-      this.abort()
-    })
-
-    // The SessionEntity class can emit a message indicating the session was paused (visibility change). This feature must stop recording if that occurs.
-    this.ee.on(SESSION_EVENTS.PAUSE, () => { this.stopRecording() })
-    // The SessionEntity class can emit a message indicating the session was resumed (visibility change). This feature must start running again (if already running) if that occurs.
-    this.ee.on(SESSION_EVENTS.RESUME, () => {
-      if (!this.initialized || this.mode === MODE.OFF) return
-      this.startRecording()
-      this.takeFullSnapshot()
-    })
-
-    // Bespoke logic for new endpoint.  This will change as downstream dependencies become solidified.
-    this.scheduler = new HarvestScheduler('blob', {
-      onFinished: this.onHarvestFinished.bind(this),
-      retryDelay: this.harvestTimeSeconds,
-      getPayload: this.prepareHarvest.bind(this),
-      raw: true
-    }, this)
-
-    // Wait for an error to be reported.  This currently is wrapped around the "Error" feature.  This is a feature-feature dependency.
-    // This was to ensure that all errors, including those on the page before load and those handled with "noticeError" are accounted for. Needs evalulation
-    registerHandler('errorAgg', (e) => {
-      this.hasError = true
-      // run once
-      if (this.mode === MODE.ERROR) {
-        this.mode = MODE.FULL
-        // if the error was noticed AFTER the recorder was already imported....
-        if (recorder && this.initialized) {
-          this.stopRecording()
-          this.startRecording()
-          this.scheduler.startTimer(this.harvestTimeSeconds)
-
-          const { session } = getRuntime(this.agentIdentifier)
-          session.state.sessionReplay = this.mode
+    const shouldSetup = (
+      getConfigurationValue(agentIdentifier, 'privacy.cookies_enabled') === true &&
+      getConfigurationValue(agentIdentifier, 'session_trace.enabled') === true
```

Comment:
Shouldn't cookies_enabled have already been checked for?

What's the intention with session_trace checking here? The config could be enabled, but it doesn't mean trace is actually running.

Reply:
Yeah these are both checked for before this module is imported.

Reply:
Not sure how well we can trust that or not, seeing as customers could import session replay agg directly themselves if they really wanted to.  Not that it matters much either way.  Will remove from agg.

Reply:
I will say, having these in the feature enables some component tests that would not be as easy if those checks are solely abstracted to the middleman layer

Reply:
can revisit later if seen as a problem, i don't see this as a blocker for now.